packages/services/server/src/public-graphql-handler.ts

```diff
@@ -0,0 +1,106 @@
+import { type FastifyBaseLogger, type RouteHandlerMethod } from 'fastify';
+import { createYoga, useExecutionCancellation, useExtendContext, useSchema } from 'graphql-yoga';
+import { useGraphQLModules } from '@envelop/graphql-modules';
+import { useHive } from '@graphql-hive/yoga';
+import { type Registry } from '@hive/api';
+import { type AuthN } from '@hive/api/modules/auth/lib/authz';
+import { cleanRequestId, TracingInstance } from '@hive/service-common';
+import { runWithAsyncContext } from '@sentry/node';
+import { asyncStorage } from './async-storage';
+import { type HiveUsageConfig } from './environment';
+import {
+  reqIdGenerate,
+  useHiveErrorHandler,
+  useHiveSentry,
+  useHiveTracing,
+  type Context,
+} from './graphql-handler';
+import { createPublicGraphQLSchema } from './public-graphql-schema';
+import { useArmor } from './use-armor';
+
+type CreatePublicGraphQLHandlerArgs = {
+  registry: Registry;
+  logger: FastifyBaseLogger;
+  authN: AuthN;
+  hiveUsageConfig: HiveUsageConfig;
+  tracing?: TracingInstance;
+};
+
+export const createPublicGraphQLHandler = (
+  args: CreatePublicGraphQLHandlerArgs,
+): RouteHandlerMethod => {
+  const publicSchema = createPublicGraphQLSchema<Context>(args.registry);
```

Comment:
I think this flow is a bit confusing when combined with our schema registry.

I saw you upgrading to a federated schema and was thinking we'd use our schema registry to do the filtering and then expose that version of the schema in this service, rather than do the filtering in this service. But this makes more sense with a federated schema where the supergraph does validation to prevent executing private queries in subgraphs...

Reply:
If we use the schema registr for filtering, we have a problem for self-hosters as they would have to retrieve the public api schema from somewhere.

We do the filtering on startup, in addition we will later use the contracts on Hive for CI/CD checks.